test/extended/dr/common.go

```diff
@@ -303,3 +311,211 @@ func checkSSH(node *corev1.Node) {
 func ssh(cmd string, node *corev1.Node) (*e2essh.Result, error) {
 	return e2essh.IssueSSHCommandWithResult(cmd, e2e.TestContext.Provider, node)
 }
+
+// installSSHKeyOnControlPlaneNodes will create a new private/public ssh keypair,
+// create a new secret for both in the openshift-etcd-operator namespace. Then it
+// will append the public key on the host core user authorized_keys file with a daemon set.
+func installSSHKeyOnControlPlaneNodes(oc *exutil.CLI) error {
+	const name = "dr-ssh"
+	const namespace = "openshift-etcd-operator"
+
+	err := createPrivatePublicSSHKeySecret(oc, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	err = createKeyInstallerDaemon(oc, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	err = ensureControlPlaneSSHAccess(oc, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ensureControlPlaneSSHAccess will test that the private key generated and installed with installSSHKeyOnControlPlaneNodes
+// is working on all control plane nodes. This effectively polls until the pod executing ssh succeeds reaching all nodes.
+func ensureControlPlaneSSHAccess(oc *exutil.CLI, name string, namespace string) error {
+	const sshPath = "/tmp/ssh"
+	const containerName = "ssh-key-tester"
+
+	var cpNodeInternalIps []string
+	for _, node := range masterNodes(oc) {
+		for _, address := range node.Status.Addresses {
+			if address.Type == corev1.NodeInternalIP {
+				cpNodeInternalIps = append(cpNodeInternalIps, address.Address)
+			}
+		}
+	}
+
+	testScript := fmt.Sprintf(`
+			#!/bin/bash	
+			SSH_BASEDIR=%s
+			P_KEY=$SSH_BASEDIR/privKey
+			# we can't change the permissions on the secret mount, thus we copy it to /root/.ssh
+			mkdir -p $HOME/.ssh
+			chmod 700 ~/.ssh
+			cp $P_KEY $HOME/.ssh/id_rsa
+			P_KEY=$HOME/.ssh/id_rsa
+			chmod 600 $P_KEY
+			
+			NODE_IPS=( %s )
+			for i in "${NODE_IPS[@]}"; do
+			  echo "testing SSH to [$i]"
+			  until ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} exit
+			  do
+				echo -n "."
+				sleep 5
+			  done 
```

Comment:
I think I was slightly confused about the echo statement but just to clarify, this keeps trying the ssh connection-and-exit until it succeeds right. Maybe a more clearer "waiting message"
```suggestion
			  until ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} exit
			  do
				echo -n "Waiting for ssh connection on ${i}"
				sleep 5
			  done 
```

Although I guess we won't ever see that message since this connection testing pod will poll forever if the keys don't get installed or ssh access isn't setup correctly for some reason.